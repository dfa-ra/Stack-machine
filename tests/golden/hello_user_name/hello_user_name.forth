
_data_
"" 31 VAR buffer
"What is your name?" VAR question
"Hello, " VAR greeting
0x80 VAR input_addr
0x84 VAR output_addr

_func_
: PRINT
    !iter
    LOOP
        1 !tmp 0 != 0
        WHILE
            3 iter - a! @
            0xFF000000 and
            dup dup !tmp
            0 = IF
            ELSE
                output_addr a! !
                iter 1 + !iter
            THEN
        REPEAT
;
: READ
    !iter1
    LOOP
        1 !tmp1 0 != 0
        WHILE
            input_addr a! @
            dup !tmp1
            iter1 a! !
            iter1 1 + !iter1
        REPEAT
;


_text_
&question PRINT

&buffer READ

&greeting PRINT
&buffer PRINT

HALT
