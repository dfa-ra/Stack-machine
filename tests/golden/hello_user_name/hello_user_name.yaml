name: hello_user_name
limit: 2000
memory_size: 400
input_streams:
  0x80: [82, 111, 109, 97, 110, 0]
  0x84: []
reports:
  - name: binary
    slice: mem
    view: |
      {instruction}
  - name: Step-by-step
    slice: all
    view: |
      tick: {cpu:tick} || microcode - {microcode} || A: {cpu:A:hex} || B: {cpu:B:hex} 
      data stack: {stack:data}
  - name: Result
    slice: last
    view: |
      numio[0x80]: {io:in:dec}
      numio[0x84]: {io:out:dec}
      tick_count = {cpu:tick}
      {condition:[io:out:str]=[What is your name?Hello, Roman]}